src/modules/session-recorder/SessionRecorder.ts

```diff
@@ -182,6 +182,15 @@ export class SessionRecorder {
   }
 
   #addEvent(eventType: RQSessionEventType, event: RQSessionEvent): void {
+    const previousSessionEvents = this.#lastTwoSessionEvents[1]?.[eventType];
+    // DOMContentLoaded events sometimes come out of order
+    if (event.type === EventType.DomContentLoaded) {
+      const insertIndex = previousSessionEvents?.findIndex((arrayEvent) => event.timestamp < arrayEvent.timestamp);
+      if (insertIndex > -1) {
+        previousSessionEvents?.splice(insertIndex, 0, event);
+        return;
+      }
+    }
     this.#lastTwoSessionEvents[1]?.[eventType]?.push(event);
```

Comment:
`previousSessionEvents.push(event)`
Also, you may check for `previousSessionEvents?.length` on top to avoid conditional checks everywhere.